test/model/message_test.dart

```diff
@@ -269,6 +271,48 @@ void main() {
     test('rendering-only update does not change timestamp (for old server versions)', () async {
       await checkRenderingOnly(legacy: true);
     });
+
+    group('Handle message edit state update', () {
```

Comment:
For all the kinds of moves, let's also test where the message is in `event.messageIds` but it isn't `event.messageId`.

Reply:
Separately, there's a different interesting kind of move that would be good to cover in these tests: a move where _both_ the stream and the topic are changed.

Reply:
> Separately, there's a different interesting kind of move that would be good to cover in these tests: a move where _both_ the stream and the topic are changed.

This particular case was covered in "message stream moved update". But I do need to cover the case where a message is edited among multiple moved message.

Reply:
Done adding the new test. See "message is both moved and updated".